### PR TITLE
Fix: correct the regular expression of getting page title from a link

### PR DESCRIPTION
**Pull request by cooltey**
When testing the PR #1695, I have noticed that some in-article links will not get the page title correctly.

Please test this PR with this `testwiki` article: https://test.wikipedia.org/wiki/Language_Variants

### CODE REVIEW

**Review by dbrant — CHANGES_REQUESTED** [review]
Looks like a good opportunity to add a couple of unit tests 😉

File: app/src/main/java/org/wikipedia/util/UriUtil.java [hunk 1]
```diff
@@ -139,13 +139,13 @@ public static String getLanguageVariantFromUri(@NonNull Uri uri) {
...
     public static String removeInternalLinkPrefix(@NonNull String link) {
-        return link.replaceFirst("/wiki/|/zh.*/", "");
+        return link.replaceFirst("/(wiki|zh.*)/", "");
     }
 
     /** For links that could be internal or external links */
     @NonNull
     private static String removeLinkPrefix(@NonNull String link) {
-        return link.replaceFirst("^.*?/wiki/", "");
+        return link.replaceFirst("^.*?/(wiki|zh.*)/", "");
```
Comment (sharvaniharan) [inline app/src/main/java/org/wikipedia/util/UriUtil.java:148]: is it ok that it now became zh specific?

File: app/src/main/java/org/wikipedia/dataclient/WikiSite.java [hunk 2]
```diff
@@ -96,6 +96,9 @@ public WikiSite(@NonNull Uri uri) {
         if (languageCode.equals(AppLanguageLookUpTable.CHINESE_LANGUAGE_CODE)) {
             languageCode = LanguageUtil.getFirstSelectedChineseVariant();
         }
+
+        // TODO: should use the default language code on the authority to prevent incorrect url such as zh-tw.wikipedia.org.
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/dataclient/WikiSite.java:100]: This can be fixed in #1695, or maybe we can ignore this because it is an edge case.

**Comment by cooltey** [comment]
> 
> 
> Looks like a good opportunity to add a couple of unit tests 😉

Done.
